DataStructure/src/main/list/ResizableArray.java

```diff
@@ -0,0 +1,167 @@
+package main.list;
+
+/**
+ * Represents a resizable array of integer elements.
+ */
+
+public class ResizableArray {
+    private int size;
+    private int firstBlankIndex;
+    private int list[];
+
+    /**
+     * Constructs a resizable array
+     * @param size
+     * size of the list
+     */
+    public ResizableArray(int size){
+        this.size = size;
+        list = new int[this.size] ;
+        firstBlankIndex = 0;
+    }
+
+    /**
+     * Constructs a resizable array with default size 50
+     */
+    public ResizableArray(){
+        size = 50;
+        list = new int[50] ;
+        firstBlankIndex = 0;
```

Comment:
Why are you initializing firstBlankIndex to 0 ? Rename this variable to writeIndex.